azavea/temperate
# Automate fixture generation

## Overview
Adds a management command `update_fixtures` to automate fixture generation and ensure fixtures are generated consistently. This will help streamline major updates to the fixtures without having to do research to recreate the current formatting.

As a consequence, this has major differences with the existing fixtures as it brings them all in line. It appears some files only change is via newlines, but ideally this will ensure that going forward all fixture changes are more meaningful going forward.

To see the PR diff without whitespace changes, you can add `?w=1` to the URL, [like so](https://github.com/azavea/temperate/pull/394/files?w=1).

## Notes
- This uses `--natural_primary` and `--natural-foreign` for all fixtures. Models that do not have natural keys defined will default to their primary key, but as natural keys are added (#357) they will take effect automatically.

## Testing Instructions
- Remove existing fixtures:
```
find . -path '*/action_steps/fixtures/*.json' -delete
find . -path '*/planit_data/fixtures/*.json' -delete
```
- Run `./scripts/manage update_fixtures`
  - Deleted fixtures should regenerate, and match files tracked in git
- Run `./scripts/update`
  - It should not error
  - You should still have the respective data

_Optional pro-mode_
- Run `docker-compose down` to ***delete all data***
- Run `./scripts/update`
  - You should have stock data again (All created data like organizations and users will still be lost)


Closes #310
